luminex/src/org/labkey/luminex/LuminexExcelParser.java

```diff
@@ -242,6 +253,93 @@ private void parseFile() throws ExperimentException
+                else
+                {
+                    crossFilePT.get(desc).put(analyteName, potentialTitrationCounts.get(desc));
+                }
+            }
+            else
+            {
+                HashMap<String, Integer> mini = new HashMap<>();
+
+                // Check for single-file titrations. Does not check for dilution counts.
+                if (potentialTitrationCounts.get(desc) != null)
+                {
+                    // To Reviewer: It's possible to make the runtime better by removing the description from consideration in the main function's potentialTitration(Raw or Summary)Counts
+                    // once we recognize that it is a titration. However I found this makes debugging a bit harder / perhaps the code less readable because the description count is
+                    // no longer there/updated and accurate to the data.
+                    mini.put(analyteName, potentialTitrationCounts.get(desc));
+                    crossFilePT.put(desc, mini);
+                    if ((type.contentEquals("raw")) && ((potentialTitrationCounts.get(desc)) >= LuminexDataHandler.MINIMUM_TITRATION_RAW_COUNT))
+                    {
+                        _titrations.put(desc, potentialTitrations.get(desc));
+                    }
+                    else if ((type.contentEquals("summary")) && ((potentialTitrationCounts.get(desc)) >= LuminexDataHandler.MINIMUM_TITRATION_SUMMARY_COUNT))
+                    {
+                        _titrations.put(desc, potentialTitrations.get(desc));
+                    }
+
+                    if (potentialTitrations.get(desc).isQcControl() )
+                    {
+                        if ((type.contentEquals("raw") && (potentialTitrationCounts.get(desc) <= LuminexDataHandler.SINGLE_POINT_CONTROL_RAW_COUNT)) ||
+                            (type.contentEquals("summary") && (potentialTitrationCounts.get(desc) == LuminexDataHandler.SINGLE_POINT_CONTROL_SUMMARY_COUNT)))
+                        {
+                            _singlePointControls.put(desc, new SinglePointControl(potentialTitrations.get(desc)));
+                        }
+                    }
+                }
+            }
+        }
+        return crossFilePT;
+    }
+
+    private void checkCrossFilePTs(Map<String, HashMap<String, Integer>> crossFilePT, Map<String, Titration> crossFilePTs, Map<String, HashSet<Double>> dilutionCounts, String type)
+    {
+        int minimum_titration_count = (type.contentEquals("raw")) ? LuminexDataHandler.MINIMUM_TITRATION_RAW_COUNT : LuminexDataHandler.MINIMUM_TITRATION_SUMMARY_COUNT;
+        int single_point_control_count = (type.contentEquals("raw")) ? LuminexDataHandler.SINGLE_POINT_CONTROL_RAW_COUNT : LuminexDataHandler.SINGLE_POINT_CONTROL_SUMMARY_COUNT;
+
+        for (String desc : crossFilePTs.keySet())
+        {
+            if (crossFilePT.containsKey(desc))
+            {
+                for (String analyte : crossFilePT.get(desc).keySet())

```

Comment:
With the refactor mentioned above to remove analyte from this map, this for loop will go away.